alica_engine/src/engine/RunningPlan.cpp

```diff
@@ -566,28 +567,29 @@ void RunningPlan::activate() {
  * Removes any robot not in robots
  * @param robots The set of robots that can participate in this running plan.
  */
-void RunningPlan::limitToRobots(
-        unordered_set<const supplementary::AgentID*, supplementary::AgentIDHash, supplementary::AgentIDEqualsComparator>
-                robots) {
+void RunningPlan::limitToRobots(const AgentSet& robots) {
     if (isBehaviour()) {
         return;
     }
     if (!_cycleManagement->mayDoUtilityCheck()) {
         return;
     }
     bool recurse = false;
-    auto curRobots = _assignment->getAllRobots();
-    for (auto& r : (*curRobots)) {
-        if (find_if(curRobots->begin(), curRobots->end(),
-                    [&r](const supplementary::AgentID* id) { return *r == *id; }) == curRobots->end()) {
-            if (_activeState != nullptr && _assignment->getRobotStateMapping()->stateOfRobot(r) == _activeState) {
+
+    // TODO: move this code into assignment to avoid the call to getAllRobots.
+    AgentSet curRobots;
+    _assignment->getAllRobots(curRobots);
+    for (const supplementary::AgentID* r : curRobots) {
+        if (find_if(robots.begin(), robots.end(), [&r](const supplementary::AgentID* id) { return *r == *id; }) ==
+                curRobots.end()) {
+            if (_activeState != nullptr && _assignment->getRobotStateMapping()->getStateOfRobot(r) == _activeState) {
                 recurse = true;
             }
             _assignment->removeRobot(r);
```

Comment:
This internally does an erase on a vector which is ineffecient. Might be better for this to be a forward_list.

Reply:
Datatypes need to be reworked, yes.
But lists are the last option normally